Actions/CheckForUpdates/yamlclass.ps1

```diff
@@ -1,163 +1,185 @@
+                    }
+                    return $true
+                }
+            }
+            if ($start.value -ne -1) {
+                $count.value = $this.content.Count-$start.value
+                return $true
+            }
+            else {
+                return $false
+            }
+        }
+    }
+
+    # Locate the lines for the specified Yaml path, given by $line and return lines as a Yaml object
+    # $start and $count are set to the start and count of the lines found
+    # Indentation of the first line returned is 0
+    # Indentation of the other lines returned is the original indentation - the original indentation of the first line
+    # Returns $null if the lines are not found
+    # See Find for more details
+    [Yaml] Get([string] $line, [ref] $start, [ref] $count) {
+        $s = 0
+        $c = 0
+        if ($this.Find($line, [ref] $s, [ref] $c)) {
+            $charCount = ($line.ToCharArray() | Where-Object {$_ -eq '/'} | Measure-Object).Count
+            [string[]] $result = @($this.content | Select-Object -Skip $s -First $c | ForEach-Object {
+                "$_$("  "*$charCount)".Substring(2*$charCount).TrimEnd()
+            } )
+            $start.value = $s
+            $count.value = $c
+            return [Yaml]::new($result)
+        }
+        else {
+            return $null
+        }
+    }
+
+    # Locate the lines for the specified Yaml path, given by $line and return lines as a Yaml object
+    # Same function as Get, but $start and $count are not set
+    [Yaml] Get([string] $line) {
+        [int]$start = 0
+        [int]$count = 0
+        return $this.Get($line, [ref] $start, [ref] $count)
+    }
+
+    # Replace the lines for the specified Yaml path, given by $line with the lines in $content
+    # If $line ends with '/', then the lines for the section are replaced only
+    # If $line doesn't end with '/', then the line + the lines for the section are replaced
+    # See Find for more details
+    [void] Replace([string] $line, [string[]] $content) {
+        [int]$start = 0
+        [int]$count = 0
+        if ($this.Find($line, [ref] $start, [ref] $count)) {
+            $charCount = ($line.ToCharArray() | Where-Object {$_ -eq '/'} | Measure-Object).Count
+            if ($charCount) {
+                $yamlContent = $content | ForEach-Object { "$("  "*$charCount)$_".TrimEnd() }
+            }
+            else {
+                $yamlContent = $content
+            }
+            $this.Remove($start, $count)
+            $this.Insert($start, $yamlContent)
+        }
+        else {
+            Write-Host -ForegroundColor Red "cannot locate $line"
+        }
+    }
+
+    # Replace all occurrences of $from with $to throughout the Yaml content
+    [void] ReplaceAll([string] $from, [string] $to) {
+        $this.content = $this.content | ForEach-Object { $_.replace($from, $to) }
+    }
+
+    # Remove lines in Yaml content
+    [void] Remove([int] $start, [int] $count) {

```

Comment:
Are the new YAML functions tested anywhere? 
Would be nice to get some test-coverage in that area.

Reply:
YamlClass is tested in CheckForUpdates.Action.Tests.ps1 and the new functions are not tested directly, but through the replace function.
Also Yamlclass is tested in End 2 end as well indirect.